### ISSUE TITLE
Add test logger

### ISSUE THREAD

**Issue by nofun97** [anz-bank/pkg#12]
Add test logger for testing the log messages of code. The API needs to be something that can be used to assert values instead of just recording the whole log. Current plan:

1. Have it be able to assert Fields
2. Assert message
3. Assert time (not important but nice to have)
4. Assert codelinker (after #10)
5. Assert level

**Comment by nofun97** [comment]
When this is fixed, feature from the PR #32 will have to be reverted. `SetOutput` creates problems as some loggers may not have an output (e.g. null logger).
